### DISCUSSION TITLE
Creating a PUT endpoint

### DISCUSSION THREAD

**Discussion by mwanago** [prisma/prisma#18298] [Q&A] [answered]
Hi. So far, I've only seen people creating `PATCH` REST endpoints with Prisma. To do this, we need to use the `update` function.
```prisma
model Article {
  id          Int     @id @default(autoincrement())
  title       String
  content     String
  description String?
}
```
```typescript
export class ArticleDto {
  title: string;
  content: string;
  description?: string;
}
```
```typescript
updateArticle(id: number, article: ArticleDto) {
  return await this.prismaService.article.update({
    data: article,
    where: {
      id,
    },
  });
}
```
If the `description` property is omitted, it is **not removed** from the article.

What's the most straightforward way to implement a `PUT` REST endpoint with Prisma? With it, omitting the `description` property would **remove it** from the article.

**Comment by nurul3101** [comment, marked as answer]
Hello @mwanago 👋 

Which database are you using? Also, am I understanding it correctly that you don't want to update specific fields but instead want to update entire database record?

**Reply by mwanago** [reply]
Hi @nurul3101. I'm using PostgreSQL. Yes, I'd like to update the entire database record. One way of doing that would be
```typescript
updateArticle(id: number, article: ArticleDto) {
  return await this.prismaService.article.update({
    data: {
      title: article.title,
      content: article.content,
      description: article.description ?? null
    },
    where: {
      id,
    },
  });
}
```
but I was wondering if there is something else built into Prisma.

**Reply by nurul3101** [reply]
In your case, I would say the only way would be to check if the description exists and set null value based on that, which you are currently doing.

There is a [set](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#set) operator which can be used to override the relation fields, but in this case you are only dealing with scalar fields, so `set` won't be helpful.
